test/cql-pytest/conftest.py

```diff
@@ -93,9 +93,14 @@ def this_dc(cql):
 # and automatically deleted at the end. We use scope="session" so that all
 # tests will reuse the same keyspace.
 @pytest.fixture(scope="session")
-def test_keyspace(cql, this_dc):
+def test_keyspace(request, cql, this_dc):
     name = unique_name()
-    cql.execute("CREATE KEYSPACE " + name + " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 1 }")
+    replication_options = [("class", "NetworkTopologyStrategy"), (this_dc, 1)]
+    if request.param == "tablets":
+        replication_options.append(("initial_tablets", 4))
+        name += '_tablets'
+    serialized_options = ", ".join([f"'{key}' : '{value}'" for key, value in replication_options])
```

Comment:
I think you over-engineered this code with all the conversions ;-)
I'm thinking ahead and I think that very soon (maybe in a few days), this "initial_tablets" thing will no longer be the way to enable tablets, but this code with the format conversions will remain.
But this is a nitpick, it's not really a terrible problem ;-)

Reply:
I think even in the post `initial_tablets` world, there will be something in the `WITH REPLICATION`, that will be used to mark the keyspace as using tablets. So maybe the "how" will change but the "where" won't (replication options).